src/plugins/intel_cpu/src/transformations/cpu_opset/arm/pass/grid_sample_decomposition.cpp

```diff
@@ -0,0 +1,813 @@
+        auto y_p1_h =
+            std::make_shared<op::v1::Multiply>(std::make_shared<op::v1::Add>(y_grid_coord, ctx.c1), height_in_float);
+        x_pixel_out = std::make_shared<op::v1::Divide>(std::make_shared<op::v1::Subtract>(x_p1_w, ctx.c1), ctx.c2);
+        y_pixel_out = std::make_shared<op::v1::Divide>(std::make_shared<op::v1::Subtract>(y_p1_h, ctx.c1), ctx.c2);
+    }
+}
+
+// Create [N, H_out, W_out] batch indices tensor for GatherND (broadcasted)
+std::shared_ptr<Node> create_batch_indices(const Ctx& ctx) {
+    auto range = std::make_shared<op::v4::Range>(ctx.i32_0, ctx.n_dim, ctx.i32_1, element::i32);
+
+    auto n_dim_1d_bs = std::make_shared<op::v0::Unsqueeze>(ctx.n_dim, ctx.i32_0);
+    auto batch_shape = std::make_shared<op::v0::Concat>(OutputVector{n_dim_1d_bs,
+                                                                     op::v0::Constant::create(element::i32, {1}, {1}),
+                                                                     op::v0::Constant::create(element::i32, {1}, {1})},
+                                                        0);
+
+    auto batch_indices = std::make_shared<op::v1::Reshape>(range, batch_shape, false);
+
+    auto n_dim_1d = std::make_shared<op::v0::Unsqueeze>(ctx.n_dim, ctx.i32_0);
+    auto h_out_1d = std::make_shared<op::v0::Unsqueeze>(ctx.h_out, ctx.i32_0);
+    auto w_out_1d = std::make_shared<op::v0::Unsqueeze>(ctx.w_out, ctx.i32_0);
+    auto bshape = std::make_shared<op::v0::Concat>(OutputVector{n_dim_1d, h_out_1d, w_out_1d}, 0);
+    auto bcast = std::make_shared<op::v3::Broadcast>(batch_indices, bshape);
+    return std::make_shared<op::v0::Convert>(bcast, element::i32);
+}
+
+// Gather from NHWC by (b, y, x)
+std::shared_ptr<Node> gather_hw_nhwc(const Ctx& ctx,
+                                     const std::shared_ptr<Node>& x_coord,
+                                     const std::shared_ptr<Node>& y_coord) {
+    auto x_i32 = std::make_shared<op::v0::Convert>(x_coord, element::i32);
+    auto y_i32 = std::make_shared<op::v0::Convert>(y_coord, element::i32);
+    auto bidx = create_batch_indices(ctx);
+
+    auto b_exp = std::make_shared<op::v0::Unsqueeze>(bidx, ctx.i32_neg1);
+    auto y_exp = std::make_shared<op::v0::Unsqueeze>(y_i32, ctx.i32_neg1);
+    auto x_exp = std::make_shared<op::v0::Unsqueeze>(x_i32, ctx.i32_neg1);
+
+    auto indices = std::make_shared<op::v0::Concat>(OutputVector{b_exp, y_exp, x_exp}, -1);
+    return std::make_shared<op::v8::GatherND>(ctx.data_nhwc, indices, 0);
+}
+
+// Reflection helpers (continuous/indexed)
+std::shared_ptr<Node> reflect_coord(const Ctx& ctx,
+                                    const std::shared_ptr<Node>& coord,
+                                    const std::shared_ptr<Node>& size_f,
+                                    bool align_corners) {
+    auto size_is_one = std::make_shared<op::v1::Equal>(size_f, ctx.c1);
+    if (align_corners) {
```

Comment:
Do we need to compute ops in `align_corners` branch if both `align_corners` and `size_is_one` are `true`? It looks like `size_m1` is always 0 in this case and the next ops are not needed, `Select` operation returns 0.